recipes-bsp/sgx/files/fix-aesm_service-cross-compilation.patch

```diff
@@ -0,0 +1,51 @@
+From 61c83346fe9e04e38334ee3cefeeb1e16d5dbb15 Mon Sep 17 00:00:00 2001
+From: Preeti Sachan <preeti.sachan@intel.com>
+Date: Fri, 1 Mar 2024 14:36:37 +0530
+Subject: [PATCH] fix aesm_service cross compilation
+
+CMake find & import native CppMicroServices library from recipe-sysroot-native
+and same is used to link for target 'aesm_service' cross build, thus throwing error:
+|~/build/tmp-*/work/corei7-64-*/sgx/2.18.1-r0/recipe-sysroot-native/usr/bin/x86_64-ese-linux/
+|../../libexec/x86_64-ese-linux/gcc/x86_64-ese-linux/11.4.0/ld:
+|~/build/tmp-*/work/corei7-64-*/sgx/2.18.1-r0/recipe-sysroot-native/usr/lib/libCppMicroServices.so.4.0.0:
+|undefined reference to `std::condition_variable::wait(std::unique_lock<std::mutex>&)@GLIBCXX_3.4.30'
+
+Explicity import & use target CppMicroServices library from recipe-sysroot to fix target build.
+
+Signed-off-by: Preeti Sachan <preeti.sachan@intel.com>
+---
+ psw/ae/aesm_service/source/core/CMakeLists.txt | 7 ++++++-
+ 1 file changed, 6 insertions(+), 1 deletion(-)
+
+diff --git a/psw/ae/aesm_service/source/core/CMakeLists.txt b/psw/ae/aesm_service/source/core/CMakeLists.txt
+index 878ba1db..09c3b403 100644
+--- a/psw/ae/aesm_service/source/core/CMakeLists.txt
++++ b/psw/ae/aesm_service/source/core/CMakeLists.txt
+@@ -9,6 +9,10 @@ if(NOT US_BUILD_SHARED_LIBS)
+   # Set up dependencies to resources from static bundles
+   usFunctionGetResourceSource(TARGET aesm_service OUT _srcs)
+ endif()
++
++set(CppMicroServices_LIBDIR $ENV{SYSROOT_LIBDIR})
++set(CppMicroServices_INCDIR $ENV{SYSROOT_INCDIR}/cppmicroservices4)
```

Comment:
Isn't find_package working for CppMicroServices? I see that it provides the cmake files for it to work.

Reply:
sgx needs both cppmicroservices-native for native binary "usResourceCompiler4" to be used during build and target cppmicroservices for cross-built library libCppMicroServices.so.
find_package imports cppmicroservices-native thus sgx binary is by default getting linked with native libCppMicroServices.so instead of target libCppMicroServices.so.